src/App.js

```diff
@@ -1,227 +1,54 @@
-    //                 done: false
-    //             }
-    //         ],
-    //         nextId: this.state.nextId + 1
-    //     });
-    // };
-
-
-    // deleteTodoHandler = (id) => {
-    //     const copiedTodos = [...this.state.todos];
-    //     const updateTodos = copiedTodos.filter(todo => {
-    //         return todo.id !== id
-    //     });
-    //     this.setState({todos: updateTodos});
-    // };
-
-
-    // toggleStatusHandler = (id) => {
-    //
-    //     //find todoItem which I want to change
-    //     const todoIndex = this.state.todos.findIndex(todo => {
-    //         return todo.id === id;
-    //     });
-    //
-    //     const updatedTodo = {
-    //         ...this.state.todos[todoIndex]
-    //     };
-    //
-    //     //change done status
-    //     const currentDoneStatus = updatedTodo.done; //false
-    //
-    //     updatedTodo.done = !currentDoneStatus; //false -> true
-    //
-    //     //save the change above safely
-    //     const todos = [...this.state.todos];
-    //     todos[todoIndex] = updatedTodo;
-    //     this.setState({todos: todos});
-    // };
-
-
     render() {
         //
-        const completedTodos = this.state.todos.filter(todo => {
+        let completedTodos = this.props.todos.filter(todo => {
             return todo.done === true;
         });
 
-        const notCompletedTodos = this.state.todos.filter(todo => {
+        let notCompletedTodos = this.props.todos.filter(todo => {
             return todo.done === false;
         });
 
-        const completedTodosNum = completedTodos.length;
-
-        const notCompletedTodosNum = notCompletedTodos.length;
-
-
         return (
 
             <BrowserRouter>
                 <div className="App">
-                    <FormContainer
-                        todos={this.state.todos}
-
-                        deleteCompletedTasksHandler={this.deleteCompletedTasksHandler}
-
-                        //CRUD method
-                        postTodoHandler={this.postTodoHandler}
-                        deleteTodoHandler={this.deleteTodoHandler}
-                        putTodoContentHandler={this.putTodoContentHandler}
-                        putTodoStatusHandler={this.putTodoStatusHandler}
-                        fetchDataHandler={this.fetchDataHandler}
-
-                        updateTodoHandler={this.updateTodoHandler}
-                        //Sort out todos
+                    <FormWrapper
+                        todos={this.props.todos}
                         completedTodos={completedTodos}
-                        completedTodosNum={completedTodosNum}
                         notCompletedTodos={notCompletedTodos}
-                        notCompletedTodosNum={notCompletedTodosNum}
-
-
                     />
                 </div>
             </BrowserRouter>
         )
     }
-
 }
+const mapStateToProps = state =>{
+    return{
+        todos: state.todos,
+        isFetching: state.isFetching
```

Comment:
Are you using `isFetching` ?

Reply:
Ahh, actually not. I looked into some articles and sample code, and it used a state "isLoading" when dispatching async code. I thought it was conventional but it is not?

Reply:
It is conventional but if you not using it, you should delete it.